packages/engine/bin/cli/src/process/local.rs

```diff
@@ -52,25 +56,25 @@ impl process::Process for LocalProcess {
 
 pub struct LocalCommand {
     engine_url: String,
-    experiment_id: String,
+    experiment_name: String,
     controller_url: String,
     max_num_workers: usize,
     output_format: OutputFormat,
 }
 
 impl LocalCommand {
     pub fn new(
-        experiment_id: &str,
+        experiment_name: ExperimentNameRef<'_>,
         max_num_workers: usize,
         controller_url: &str,
         output_format: OutputFormat,
     ) -> Result<Self> {
         // The NNG URL that the engine process will listen on
-        let engine_url = format!("ipc://run-{experiment_id}");
+        let engine_url = format!("ipc://run-{experiment_name}"); // TODO: is this okay?
```

Comment:
This might be a problem. We previously had a usually unique '`experiment_id`' that was generated by the name + some rng. 

Perhaps this is an indicator that experiment UUID generation should be done outside of the engine

Reply:
Yeah, UUID generation could be moved from `new` (`src/config/experiment.rs:23`) to `run_experiment_with_manifest` (`bin/cli/src/experiment.rs:68`) and then the UUID added to `init_message` in `run_experiment_with_manifest`.

Reply:
Do we wish to make the change in this PR or create a follow-up task?

Reply:
Might leave this as draft for a while and wait until we have some automated integration tests in place, just because this affects a lot of files.